lib/checks/lists/only-listitems.js

```diff
@@ -1,18 +1,41 @@
 var bad = [],
 	permitted = ['STYLE', 'META', 'LINK', 'MAP', 'AREA', 'SCRIPT', 'DATALIST', 'TEMPLATE'],
-	hasNonEmptyTextNode = false;
+	hasNonEmptyTextNode = false,
+	hasListItem = false,
+	liItemsWithRole = 0,
+	isEmpty = true;
 
 virtualNode.children.forEach(({ actualNode }) => {
 	var nodeName = actualNode.nodeName.toUpperCase();
-	if (actualNode.nodeType === 1 && nodeName !== 'LI' && permitted.indexOf(nodeName) === -1) {
-		bad.push(actualNode);
+	var isListItemRole = false;
+
+	if (actualNode.nodeType === 1 && permitted.indexOf(nodeName) === -1) {
+		var role = (actualNode.getAttribute('role') || '').toLowerCase();
+		isListItemRole = role === 'listitem' || (nodeName === 'LI' && !role);
+		hasListItem = hasListItem || (nodeName === 'LI' && isListItemRole) || isListItemRole;
```

Comment:
It also seems a little redundant: 
```hasListItem = hasListItem || (nodeName === 'LI' && isListItemRole) || isListItemRole;```
The code checks the nodeName for 'LI' and a role more than once, and then checks against itself again a few times. I wonder if we can simplify.